applications/ShallowWaterApplication/custom_elements/wave_element.h

```diff
@@ -335,11 +335,11 @@ class WaveElement : public Element
         const std::size_t I,
         const std::size_t J);
 
-    static const array_1d<double,3> VectorProduct(
+    static array_1d<double,3> VectorProduct(
```

Comment:
I'd have done this method `const`.

Reply:
I do not really understand all the stuff related to const. Correct me if I'm wrong:
`static const array_1d VectorProduct(...);` -> The output is declared const. This is ignored by the compiler, the CI throws a warning
`static array_1d VectorProduct(...) const;` -> A static method can't be marked as const (it does not have the `this` pointer). Compilation error.

Reply:
About 1, this is ignored as const must be declared outside (when assigning the return value). About 2, sorry, I didn't note these being static. Forget about my comment.